### PR TITLE
 [Storage][Pruner] Refactor pruners.

### PR DESCRIPTION
**Pull request by grao1991**
### Description

- Removed unnecessary functions from traits.
- Share the PrunerWorker struct, and use it to manage the threads.
- Manage min_readable_version in the pruner manager for external calls.

### Test Plan
Existing UTs.

### CODE REVIEW

File: storage/aptosdb/src/pruner/pruner_manager.rs [hunk 1]
```diff
@@ -10,25 +10,30 @@ use aptos_types::transaction::Version;
...
+    }
 
     fn get_min_readable_version(&self) -> Version;
 
     /// Sets pruner target version when necessary.
     fn maybe_set_pruner_target_db_version(&self, latest_version: Version);
 
-    fn set_pruner_target_db_version(&self, latest_version: Version);
+    // Only used at the end of fast sync to store the min_readable_version to db and update the
+    // in memory progress.
```
Comment (msmouse) [inline storage/aptosdb/src/pruner/pruner_manager.rs:33]: 👍

File: storage/aptosdb/src/pruner/pruner_metadata.rs [hunk 2]
```diff
@@ -1,22 +0,0 @@
-// Copyright © Aptos Foundation
```
Comment (msmouse) [inline storage/aptosdb/src/pruner/pruner_metadata.rs:1]: 😲

File: storage/aptosdb/src/pruner/pruner_worker.rs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        while !self.quit_worker.load(Ordering::SeqCst) {
+            let pruner_result = self.pruner.prune(self.batch_size);
+            if pruner_result.is_err() {
+                sample!(
+                    SampleRate::Duration(Duration::from_secs(1)),
+                    error!(error = ?pruner_result.err().unwrap(),
+                        "Pruner has error.")
+                );
+                sleep(Duration::from_millis(self.pruning_time_interval_in_ms));
+                return;
```
Comment (msmouse) [inline storage/aptosdb/src/pruner/pruner_worker.rs:63]: ? if you return, how is `sample!()` useful?

**Review by msmouse — APPROVED** [review]
Beautiful. Do you plan to extract a common Manager as well? 😂

File: storage/aptosdb/src/pruner/pruner_manager.rs (hunk 1 above)
Comment (grao1991) [reply storage/aptosdb/src/pruner/pruner_manager.rs:33]: > Beautiful. Do you plan to extract a common Manager as well? 😂

Was planning to do it, but I'm lazy, next time..

**Review by zekun000 — APPROVED** [review]

**Pull request merged**